### PR TITLE
feat(typewriter): Add docTags map to DocsSpec interface.

### PR DESCRIPTION
**Pull request by mikewrighton**
Add `docTags` field to the DocSpec interface.

This is a dependency of https://github.com/aws/aws-cdk/issues/26839.

### CODE REVIEW

File: packages/@cdklabs/typewriter/src/documented.ts [hunk 1]
```diff
@@ -85,6 +85,12 @@ export interface DocsSpec {
    * @default none
    */
   see?: string;
+  /**
+   * The `@cloudformationResource` string for Cfn resource classes
+   *
+   * @default none
+   */
+  cloudformationResource?: string;
```
Comment (mrgrain) [inline packages/@cdklabs/typewriter/src/documented.ts:93]: `@cdklabs/typewriter` is intended as a general purpose tool. The other doc flags are all universally applicable and/or inspired by jsii which also aims to be a universal tool..

I think we need a generic feature to support arbitrary additional doctags.

**Review by mrgrain — DISMISSED** [review]

**Comment by mrgrain** [comment]
The PR title also needs some tweaking. Different scope and in the context of this repo, I'd say this will be a feature.

File: packages/@cdklabs/typewriter/src/documented.ts [hunk 2]
```diff
@@ -85,6 +85,12 @@ export interface DocsSpec {
    * @default none
    */
   see?: string;
+  /**
+   * A map of document tags where the key represents the tag name.
+   *
+   * @default none
+   */
+  docTags?: Map<string, string>;
```
Comment (rix0rrr) [inline packages/@cdklabs/typewriter/src/documented.ts:93]: Generally we'd do a `Record<string, string>` here I think.

**Review by rix0rrr — APPROVED** [review]
Conditional approval (I made it a Draft). Address my comment or not (your choice), and then make it a proper PR again to merge at your leisure.

**Pull request merged**